src/main/java/com/openlattice/mapstores/TestDataFactory.java

```diff
@@ -154,6 +146,29 @@ public static Principal rolePrincipal() {
         return new Principal( PrincipalType.ROLE, randomAlphanumeric( 5 ) );
     }
 
+    public static SecurablePrincipal securableUserPrincipal() {
+        return securablePrincipal( PrincipalType.USER );
+    }
+
+    public static SecurablePrincipal securablePrincipal( PrincipalType type ) {
+        // TODO after Java 13: use switch expression
+        Principal principal;
+        if ( type == PrincipalType.ROLE ) {
```

Comment:
why are we waiting for java 13 to use a switch statement?

Reply:
We can use switch, but not as an expression. In Java 13, we will be able to do this:
```
        var principal = switch ( type ) {
            case ( PrincipalType.ROLE ):
                rolePrincipal();
                break;
            case ( PrincipalType.ORGANIZATION ):
                organizationPrincipal();
                break;
            default:
                userPrincipal();
        }
```

Reply:
https://openjdk.java.net/jeps/325

Reply:
I see. I think it would still be preferable to use a switch statement that assigns `principal` a value though, just for clarity and in case we end up adding additional principal types. Not hugely important though!